platform/ios/src/MLNMapView.mm

```diff
@@ -552,6 +568,12 @@ - (nonnull NSURL *)styleURL
         return self.residualStyleURL;
     }
 
+    // If style was initialized with JSON, return nil
+    if (self.mbglMap.getStyle().getJSON().length() > 0 &&
+        self.mbglMap.getStyle().getURL().empty()) {
+        return nil;
```

Comment:
The return type is marked with `nonnull` and the interface with `null_resettable`, which means this getter should not return null if I understand it correctly. We should not modify this, because I think it would change the Swift API.

Maybe you can return `[NSURL URLWithString:@"local://style.json"]` instead?